### ISSUE TITLE
How to recover jobs?

### ISSUE THREAD

**Issue by jon-malave** [node-cron/node-cron#209]
So while node-cron seems to be working fairly well on my end so far, I'm curious about how best to maintain the state of the jobs and recover them? Say in the event of a server crash or reboot requirement. While it's not exactly a Bug related to node-cron, I think maintain the state of jobs and recovery is an important feature.

**Comment by TDola** [comment]
Ideally whatever you are running should be smart enough to keep a record of the last time it ran and process everything spanning that time.

**Comment by jon-malave** [comment]
> Ideally whatever you are running should be smart enough to keep a record of the last time it ran and process everything spanning that time.

I was able to code up a solution using firestore as my backend to maintain a record of each scheduled job. If server reboots, I implemented a restore mode where I perform some logic on the records pulled back in from firestore to determine which should be restored or not based on timestamps and such.

**Comment by TDola** [comment]
Yep basically that's what I do too.  I am working on a way to scan forwards and backwards and highlight to a user expected schedules.  Then knowing your own down time, you can visually see how many times the schedule was missed, and decide to run it manually.
